library/core/src/slice/mod.rs

```diff
@@ -2031,6 +2034,256 @@ impl<T> [T] {
         unsafe { (from_raw_parts_mut(ptr, mid), from_raw_parts_mut(ptr.add(mid), len - mid)) }
     }
 
+    /// Divides one slice into two at an index returning, `None` if slice is too
+    /// short.
+    ///
+    /// The first will contain all indices from `[0, mid)` (excluding
```

Comment:
This paragraph is copied from `split_at`. But unlike there, here there is too much separation between _"Divides one slice into two"_ and _"The first"_ to be immediately obvious what first it's referring to. I think you should find a way to reword this.